SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h

```diff
@@ -24,6 +24,27 @@ class LHEEventProduct {
 	LHEEventProduct(const lhef::HEPEUP &hepeup,
 			const double originalXWGTUP) : 
 	  hepeup_(hepeup), originalXWGTUP_(originalXWGTUP) {}
+	LHEEventProduct(LHEEventProduct&& other) {
```

Comment:
may I ask what is addition used for?

Reply:
`LHEEventProduct(LHEEventProduct&& other)` is a move constructor and `LHEEventProduct& operator=(LHEEventProduct&& other)` is a move assignment operator. They need to be explicitly defined for `LHEEventProduct` because of `auto_ptr` member.

Motivation (from PR description) is to replace the current pre-C++11 "swap-or-copy" idiom inside `edm::Wrapper` with move in order to not require move-only products to define `swap()` when compiler-generated move constructor and assignment operator are enough.